### PR TITLE
KMP algorithm in dart added

### PR DESCRIPTION
**Pull request by disha2sinha**
Fixes #2555 

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Brief Description added
-  Added Line between Functions
-  Added line at the end of file

#### Languages Used:
-  Dart

#### Files Added:
-  KMP.Dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Knuth_Morris_Pratt_Algorithm/KMP.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+/*
+Enter Text:
+gfmghdmgmgcc
+Enter Pattern:
+mg
+Pattern found at position: 3
+Pattern found at position: 7
+Pattern found at position: 9 */
```
Comment (singh-shreya6) [inline Knuth_Morris_Pratt_Algorithm/KMP.dart:67]: Please add a blank line at end of file.

File: Knuth_Morris_Pratt_Algorithm/KMP.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        }
+        if (pattern[k + 1] == pattern[j])
+        {
+            k = k + 1;
+        }
+        pi.insert(j,k);
+    }
+    return pi;
+}
+void KMP_matcher (String text, String pattern)
```
Comment (singh-shreya6) [inline Knuth_Morris_Pratt_Algorithm/KMP.dart:23]: Please add a blank line between functions.

File: Knuth_Morris_Pratt_Algorithm/KMP.dart [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import 'dart:io';
```
Comment (singh-shreya6) [inline Knuth_Morris_Pratt_Algorithm/KMP.dart:1]: Please add a brief problem description.

**Comment by singh-shreya6** [comment]
Please complete the PR template.

**Comment by disha2sinha** [comment]
Changes made

File: Knuth_Morris_Pratt_Algorithm/KMP.dart [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
```
Comment (singh-shreya6) [inline Knuth_Morris_Pratt_Algorithm/KMP.dart:1]: Please add the import statement after comment

**Comment by disha2sinha** [comment]
> 
> 
> Post that one change, you can squash your commits.

done

**Pull request merged**